src/server/service/kv.rs

```diff
@@ -140,16 +145,43 @@ impl<T: RaftStoreRouter + 'static> Service<T> {
         let status = RpcStatus::new(code, Some(format!("{}", err)));
         ctx.spawn(sink.fail(status).map_err(|_| ()));
     }
+
+    fn send_fail_status_to_stream<M>(
+        &self,
+        ctx: RpcContext,
+        sink: ServerStreamingSink<M>,
+        err: Error,
+        code: RpcStatusCode,
+    ) {
+        let status = RpcStatus::new(code, Some(format!("{}", err)));
+        ctx.spawn(sink.fail(status).map_err(|_| ()));
+    }
 }
 
-fn make_callback<T: Debug + Send + 'static>() -> (Box<FnBox(T) + Send>, oneshot::Receiver<T>) {
+fn make_callback<T: Send + Debug + 'static>() -> (Box<FnBox(T) + Send>, oneshot::Receiver<T>) {
     let (tx, rx) = oneshot::channel();
-    let callback = move |resp| {
-        tx.send(resp).unwrap();
-    };
+    let callback = move |resp| tx.send(resp).unwrap();
     (box callback, rx)
 }
 
+pub fn make_stream_callback<T>(channel_size: usize) -> (OnResponse<T>, mpsc::Receiver<T>)
```

Comment:
I think it is not consistent for `make_stream_callback` returns a `(OnResponse, rx)` while `make_callback` returns a `(Fn, rx)`.

Reply:
In addition, I think it is not reasonable for `make_....callback` actually makes a `OnResponse`.